### DISCUSSION TITLE
Deprecate v1 sdk

### DISCUSSION THREAD

**Discussion by patrickjamesbarry** [aws/aws-sdk-java-v2#5088] [Migration from v1 to v2]
Amazon has announced the end of support for aws sdk v1.  Why do they not deprecate the classes and add javadocs letting devs know what new class to use instead?

**Comment by debora-ito** [comment]
We are adding **Migrating to AWS SDK for Java v2** information to the top description of some handwritten classes, like [RequestHandler2](https://docs.aws.amazon.com/AWSJavaSDK/latest/javadoc/com/amazonaws/handlers/RequestHandler2.html) and [EC2MetadataUtils](https://docs.aws.amazon.com/AWSJavaSDK/latest/javadoc/com/amazonaws/util/EC2MetadataUtils.html).

For a more granular v1-to-v2 mapping, including API differences and behavior changes, this information will be added to the **Migrate to version 2** section of the Developer Guide - for example:
-  [Credentials provider changes
](https://docs.aws.amazon.com/sdk-for-java/latest/developer-guide/migration-client-credentials.html)
- [S3 TransferManager](https://docs.aws.amazon.com/sdk-for-java/latest/developer-guide/migration-transfer-manager.html)
- [IAM Policy Builder](https://docs.aws.amazon.com/sdk-for-java/latest/developer-guide/migration-iam-policy-builder.html)
- and [more](https://docs.aws.amazon.com/sdk-for-java/latest/developer-guide/migration-whats-different.html)

The "Migrate to version 2" documentation is a work in progress, we are actively working on it and constantly adding new pages. We understand DynamoDB Enhanced client is missing right now, and it's a highly needed documentation.

I don't think we'll be able to add the v2 equivalent classes to all the v1 Javadocs, because most v1 classes are auto-generated, and the logic to find the v2 equivalent is not always straightforward.

**Comment by patrickjamesbarry** [comment]
It is crazy that aws sdk v2 was released in 2018 and migration guides are still a work in progress in 2024. If aws devs find the logic hard to find the v2 equivalent of classes, imagine how a consumer of the api feels!  

It is always nice to visually see and be warned when you are using a deprecated class in your IDE and for the javadoc help guide you to its replacement. Was just trying to throw out a suggestion that would help your consumers.
